include/SFML/Config.hpp

```diff
@@ -178,25 +180,20 @@ namespace sf
     // we can use them without doing any kind of check
 
     // 8 bits integer types
-    typedef signed   char Int8;
-    typedef unsigned char Uint8;
+    using Int8 = std::int_least8_t;
+    using Uint8 = std::uint_least8_t;
 
     // 16 bits integer types
-    typedef signed   short Int16;
-    typedef unsigned short Uint16;
+    using Int16 = std::int_least16_t;
+    using Uint16 = std::uint_least16_t;
 
     // 32 bits integer types
-    typedef signed   int Int32;
-    typedef unsigned int Uint32;
+    using Int32 = std::int_least32_t;
+    using Uint32 = std::uint_least32_t;
 
     // 64 bits integer types
-    #if defined(_MSC_VER)
-        typedef signed   __int64 Int64;
-        typedef unsigned __int64 Uint64;
-    #else
-        typedef signed   long long Int64;
-        typedef unsigned long long Uint64;
-    #endif
+    using Int64 = std::int_least64_t;
+    using Uint64 = std::uint_least64_t;
```

Comment:
I feel like this "core" change, would need to be discussed in a separate PR.
One might even go as far as argue, that the custom `sf::*` types are no longer needed.